src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
+      if (cameraRef.current) {
+        cameraRef.current.setCamera({
+          centerCoordinate: [longitude, latitude],
+          zoomLevel: zoom,
+          animationDuration: 1000
+        });
+      }
+
+      // Wait for the animation to finish
+      setTimeout(() => {
+        setIsAnimating(false);
+      }, 1000);
+    },
+    [isAnimating, zoomLevel]
+  );
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
-        onPress={onMapPress}
+      <LoadingSpinnerMap loading={isLoading} />
+      <MapLibreGL.MapView
+        style={mapStyle || styles.map}
+        styleJSON="https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json"
+        onPress={handleMapPress}
         ref={refForMapView}
-        rotateEnabled={false}
-        showsBuildings={false}
-        showsPointsOfInterest={false}
-        showsUserLocation={showsUserLocation}
-        showsMyLocationButton={false}
-        toolbarEnabled={false}
-        style={[stylesForMap().map, mapStyle]}
-        userLocationPriority="balanced"
-        minZoom={minZoom}
-        mapPadding={{
-          top: 0,
-          right: 0,
-          bottom: -normalize(30),
-          left: 0
-        }}
-        legalLabelInsets={{
-          top: 0,
-          right: 0,
-          bottom: normalize(10),
-          left: 0
+        onRegionDidChange={(region) => {
+          const newZoomLevel = Math.round(region.properties.zoomLevel);
+          if (!isNaN(newZoomLevel)) {
+            setZoomLevel(newZoomLevel);
+          }
         }}
+        showUserLocation={showsUserLocationSetting}
       >
-        <UrlTile
-          urlTemplate="https://tileserver-gl.smart-village.app/styles/osm-liberty/{z}/{x}/{y}.png"
-          shouldReplaceMapContent={device.platform === 'ios'}
-        />
-        {!!geometryTourData?.length && (
-          <Polyline
-            coordinates={geometryTourData}
-            strokeWidth={2}
-            strokeColor={colors.primary}
-            zIndex={1}
-          />
-        )}
-        {locations?.map((marker, index) => {
+        <MapLibreGL.Camera ref={cameraRef} minZoomLevel={minZoom} />
+
+        {clusters.map((cluster, index) => {
+          const [longitude, latitude] = cluster.geometry.coordinates;
+          const { cluster: isCluster, point_count: pointCount } = cluster.properties;
+
+          if (clusteringEnabled && isCluster) {
+            return renderCluster({
+              clusterColor: colors.primary,
+              geometry: { coordinates: [longitude, latitude] },
+              id: `cluster-${index}`,
```

Comment:
setting this to `cluster-3` results in `cluser-cluster-3` in the method, right? i guess it is not intended?

Reply:
you are right, this was not intended.
it will be changed in the upcoming commit to: 
`id: index`